mobile/integration_test/test_utils/general_helper.dart

```diff
@@ -62,3 +63,21 @@ void immichWidgetTest(
     semanticsEnabled: false,
   );
 }
+
+Future<void> pumpUntilFound(
+    WidgetTester tester,
+    Finder finder, {
+      Duration timeout = const Duration(seconds: 120),
+    }) async {
+  bool timerDone = false;
+  final timer = Timer(timeout, () => throw TimeoutException("Pump until has timed out"));
+  while (timerDone != true) {
+    await tester.pump();
+
+    final found = tester.any(finder);
+    if (found) {
```

Comment:
I had to think about this for a while.
What about

```dart
final found = false;
//...
while (!found) {
//...
found = tester.any(finder);
//...
}

```

Reply:
I took this code from somewhere else, I am not sure refactoring it in this way really makes it any more or less readable. Would rather leave it alone given it works perfectly :smile:

Reply:
My concern is just the naming here. Because `timerDone` is the opposite of what this variable is actually doing :D

Reply:
Yea true, reading it again I think we can just drop the variable. Will quickly test now